bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java

```diff
@@ -662,6 +662,13 @@ private void rangeCheck(long index, int size) {
         }
     }
 
+    private void fillerValueCheck(int index, int size) {
+        if (!TypeChecker.hasImplicitInitialValue(elementType) && (index > size)) {
```

Comment:
`(index > size)` this is the consecutive insertion check right?
Shall we extract a variable or add a comment explaining this non consecutive add rule.